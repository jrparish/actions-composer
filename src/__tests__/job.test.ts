import { describe, it, expect } from 'vitest';

import { Job } from '../job';

describe('Job', () => {
  it('should create a job', () => {
    const job = new Job('jobKey', {
      runsOn: 'ubuntu-latest',
      continueOnError: true,
      timeoutMinutes: 10,
      strategy: {
        fastFail: true,
        maxParallel: 11
      },
      steps: [
        {
          name: 'step',
          continueOnError: false,
          timeoutMinutes: 5,
          workingDirectory: '~/'
        },
        {
          name: 'External action',
          uses: 'actions/checkout@v2',
          with: {
            stringValue: 'string',
            numberValue: 10,
            booleanValue: false
          }
        }
      ]
    });
    expect(job.toAction()).toMatchSnapshot();
  });
});
