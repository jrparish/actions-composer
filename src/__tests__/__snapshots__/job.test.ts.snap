// Vitest Snapshot v1

exports[`Job > should create a job 1`] = `
{
  "container": undefined,
  "continue-on-error": true,
  "defaults": undefined,
  "env": undefined,
  "id": undefined,
  "if": undefined,
  "name": undefined,
  "needs": undefined,
  "outputs": undefined,
  "runs-on": "ubuntu-latest",
  "services": undefined,
  "steps": [
    {
      "continue-on-error": false,
      "name": "step",
      "timeout-minutes": 5,
      "working-directory": "~/",
    },
    {
      "name": "External action",
      "uses": "actions/checkout@v2",
      "with": {
        "booleanValue": false,
        "numberValue": 10,
        "stringValue": "string",
      },
    },
  ],
  "strategy": {
    "fail-fast": true,
    "max-parallel": 11,
  },
  "timeout-minutes": 10,
}
`;
